### DISCUSSION TITLE
Updating from 32-bit to 64-bit: preserve settings?

### DISCUSSION THREAD

**Discussion by UTANdev** [WinMerge/winmerge#1385] [Q&A] [answered]
I just noticed that I'd been using the 32-bit version of WinMerge and there's a shiny 64-bit version available. How do I ensure that my settings get migrated over when I install the 64-bit version, and that they don't get deleted when I uninstall the 32-bit version?

**Comment by sdottaka** [comment, marked as answer]
When WinMerge is uninstalled, the settings are deleted.
For this reason, please do the following (You may be asked to restart your computer)

1. press Windows+R and type regedit
2. rename `WinMerge` in the registry key `Computer\HKEY_CURRENT_USER\Software\Thingamahoochie\WinMerge` to `_WinMerge` etc. 

![image](https://user-images.githubusercontent.com/98126/176559181-ce4fcdc2-656f-42f6-87f6-2a092ff7affa.png)

3. Install the 64-bit version of WinMerge. 
4. If your computer has not been restarted, press F5 to update to the latest state. If your computer has been rebooted, run regedit again.
5. delete the registry key `Computer\HKEY_CURRENT_USER\Software\Thingamahoochie\WinMerge
6. restore the registry key renamed in step 2.

**Comment by UTANdev** [comment]
One detail: if you uninstall a different class (32- vs. 64-bit) of WinMerge than the one you install, you must edit the `Computer\HKEY_CURRENT_USER\SOFTWARE\Thingamahoochie\WinMerge\Executable` key to reflect the true location of the executable.

For example, if upgrading from 32-bit to 64-bit WinMerge:

- After renaming `WinMerge` to `_WinMerge`, change the `Computer\HKEY_CURRENT_USER\SOFTWARE\Thingamahoochie\_WinMerge\Executable` key's value from `C:\Program Files (x86)\WinMerge\WinMergeU.exe` to `C:\Program Files\WinMerge\WinMergeU.exe`.
- Uninstall the old version.
- Install the new version.
- Delete the new `Computer\HKEY_CURRENT_USER\SOFTWARE\Thingamahoochie\WinMerge` tree.
- Rename `Computer\HKEY_CURRENT_USER\SOFTWARE\Thingamahoochie\_WinMerge` back to `Computer\HKEY_CURRENT_USER\SOFTWARE\Thingamahoochie\WinMerge`.
